### PR TITLE
ZFIN-8949 convert ExpressionExperiment to ExpressionExperiment2 class

### PR DESCRIPTION
**Pull request by cmpich**

### CODE REVIEW

File: source/org/zfin/framework/presentation/tags/CreateLinkTag.java [hunk 1]
```diff
@@ -182,7 +182,7 @@ else if (o instanceof Term) {
         else if (o instanceof TermDTO)
             link = TermDTOPresentation.getLink((TermDTO) o);
         else if (o instanceof ExpressionResult)
```
Comment (rtaylorzfin) [inline source/org/zfin/framework/presentation/tags/CreateLinkTag.java:184]: Seems like this line should also be changed to ExpressionResult2.  Could even use the new syntax (https://blogs.oracle.com/javamagazine/post/pattern-matching-for-instanceof-in-java-14)

File: source/org/zfin/expression/repository/HibernateExpressionRepository.java [hunk 2]
```diff
@@ -59,1296 +59,1283 @@
...
+            AND expressionFigureStage.startStage = :start
+            AND expressionFigureStage.endStage = :end
+            AND superTerm = :superTerm
+            AND expressionFound = :expFound
+            """, ExpressionResult2.class);
+        query.setParameter("expressionExperiment", result.getExpressionFigureStage().getExpressionExperiment());
+        query.setParameter("start", result.getExpressionFigureStage().getStartStage());
+        query.setParameter("end", result.getExpressionFigureStage().getEndStage());
+        query.setParameter("superTerm", unspecified);
+        query.setParameter("expFound", result.isExpressionFound());
```
Comment (rtaylorzfin) [inline source/org/zfin/expression/repository/HibernateExpressionRepository.java:1173]: I'm sure you've worked out the logic here, but I'm not sure why the `"expressionFound", true` gets changed to `"expFound", result.isExpressionFound()`

**Review by rtaylorzfin — APPROVED** [review]
Looks good

File: source/org/zfin/expression/repository/HibernateExpressionRepository.java (hunk 2 above)
Comment (cmpich) [reply source/org/zfin/expression/repository/HibernateExpressionRepository.java:1173]: Thanks for this catch. I will undo this change from the original one.

**Pull request merged**